packages/clerk-js/src/core/resources/Session.ts

```diff
@@ -160,7 +160,11 @@ export class Session extends BaseResource implements SessionResource {
     const cachedEntry = skipCache ? undefined : SessionTokenCache.get({ tokenId }, leewayInSeconds);
 
     if (cachedEntry) {
-      return cachedEntry.tokenResolver.then(res => res.getRawString());
+      const cachedToken = await cachedEntry.tokenResolver.then(res => res);
+      if (!template) {
+        eventBus.dispatch(events.TokenUpdate, { token: cachedToken });
+      }
+      return cachedToken.getRawString();
```

Comment:
🤔 This means we forgot to apply Bryce change to the beta version?
ref: https://github.com/clerk/javascript/pull/2585

Reply:
We did not "forget" per se, as it was not necessarily needed but I decided to port it for two reasons:
- the vendor-level protection will be around for ~a year so there's a small chance that someone upgrading to v5 might break some of their users after deploying to a vendor that still has the protections active
- its going to be easier for us to debug any related issues that come up if both versions share the same implementation while people are upgrading